### PR TITLE
feat: RBTC - BTCS bridge

### PR DESCRIPTION
**Pull request by creed-victor**
AC:
- user can deposit RBTC to RSK side and receive BTCS on Binance Smart-Chain side
- user can deposit BTCS to Binance Smart-Chain side and receive RBTC on RSK side

### CODE REVIEW

File: src/app/containers/BridgePage/components/DestinationChainCard/index.tsx [hunk 1]
```diff
@@ -226,6 +210,36 @@ export function DestinationChainCard() {
...
+                    : receiver.value.toLowerCase()}
+                </div>
+              </>
+            )}
+            {receiveAtExternalWallet && (
+              <Input
+                className="mb-1"
+                value={receiver.value}
+                onChange={value => receiver.set((value || '').toLowerCase())}
+                placeholder="Enter or paste address"
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/DestinationChainCard/index.tsx:228]:
```suggestion
                placeholder="Enter or paste an address"
```

File: src/app/containers/BridgePage/components/DestinationChainCard/index.tsx [hunk 2]
```diff
@@ -226,6 +210,36 @@ export function DestinationChainCard() {
...
+            <button
+              className="text-secondary font-medium hover:underline focus:outline-none"
+              onClick={() => {
+                setReceiveAtExternalWallet(!receiveAtExternalWallet);
+                receiver.set('');
+              }}
+            >
+              {receiveAtExternalWallet
+                ? '- Click here to receive at current address'
+                : '+ Click here to receive at external address'}
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/DestinationChainCard/index.tsx:240]:
```suggestion
                ? '- Click here to receive at the current address'
                : '+ Click here to receive at an external address'}
```

File: src/app/containers/BridgePage/components/TransactionDialog/components/StepConfirm.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/TransactionDialog/components/StepConfirm.tsx:1]: This is the same component as `StepApprove`, delete one of them and create an interface that will have an enum (e.g. `Step` with values `Approve` and `Confirm`).

File: src/app/containers/BridgePage/components/TransactionDialog/components/StepMain.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Spinner } from '@blueprintjs/core';
+
+export const StepMain: React.FC = () => (
+  <>
+    <Spinner size={96} />
+  </>
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/TransactionDialog/components/StepMain.tsx:7]:
```suggestion
    <Spinner size={96} />
```

File: src/app/containers/BridgePage/components/TransactionDialog/components/StepPending.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/TransactionDialog/components/StepPending.tsx:1]: This is the same component as `StepConfirmed`, delete one of them and unify them.

File: src/app/containers/BridgePage/components/TransactionDialog/components/WalletLogo.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { getWalletImage, getWalletName } from '../utils';
+import { WalletLogoContainer, WalletLogoImage } from '../styled';
+
+interface IWalletLogoProps {
+  wallet: string;
+}
+
+export const WalletLogo: React.FC<IWalletLogoProps> = ({ wallet }) => {
+  return (
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/TransactionDialog/components/WalletLogo.tsx:10]:
```suggestion
export const WalletLogo: React.FC<IWalletLogoProps> = ({ wallet }) => (
```

File: src/app/containers/BridgePage/components/TransactionDialog/utils.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import MetamaskLogo from 'assets/wallets/metamask.svg';
+import NiftyLogo from 'assets/wallets/nifty.png';
+import LiqualityLogo from 'assets/wallets/liquality.svg';
+import PortisLogo from 'assets/wallets/portis.svg';
+import LedgerLogo from 'assets/wallets/ledger.svg';
+import TrezorLogo from 'assets/wallets/trezor.svg';
+import WalletConnectLogo from 'assets/wallets/walletconnect.svg';
+
+export const getWalletName = walletName => {
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/TransactionDialog/utils.ts:9]: Rewrite to `switch`.

File: src/app/containers/BridgePage/components/TransactionDialog/utils.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  if (walletName === 'liquality') return 'Liquality';
+  if (walletName === 'nifty') return 'Nifty';
+  if (walletName === 'portis') return 'Portis';
+  if (walletName === 'ledger') return 'Ledger';
+  if (walletName === 'trezor') return 'Trezor';
+  if (walletName === 'wallet-connect') return 'Wallet Connect';
+  return 'MetaMask';
+};
+
+export const getWalletImage = walletName => {
```
Comment (tiltom) [inline src/app/containers/BridgePage/components/TransactionDialog/utils.ts:19]: Rewrite to `switch`.

File: src/app/containers/BridgePage/saga.ts [hunk 9]
```diff
@@ -181,8 +184,34 @@ function* confirmTransfer() {
       ).toLowerCase();
 
       if (
+        payload.form.sourceNetwork === NetworkType.BSC &&
+        asset?.aggregatorData.aggregatorContractAddress &&
+        asset?.aggregatorData.redeem ===
+          BabelFishRedeem.REDEEM_WITH_EXTRA_DATA &&
+        asset?.aggregatorData.customBridge
+      ) {
+        let basset = (
```
Comment (tiltom) [inline src/app/containers/BridgePage/saga.ts:193]:
```suggestion
        let bAsset = (
```

File: src/styles/tailwindcss/components/dialog.css [hunk 10]
```diff
@@ -79,8 +80,8 @@
 
 
     .dialog h1, .dialog .title {
-        @apply font-semibold mb-12 text-center mx-auto;
-        max-width: 240px;
+        @apply font-semibold text-center mx-auto;
+        margin-bottom: 40px;
```
Comment (tiltom) [inline src/styles/tailwindcss/components/dialog.css:84]:
```suggestion
        @apply font-semibold text-center mx-auto mb-10;
```

File: src/styles/tailwindcss/components/dialog.css [hunk 11]
```diff
@@ -56,8 +56,9 @@
         @apply bg-black;
         @apply text-white;
         @apply w-full;
```
Comment (tiltom) [inline src/styles/tailwindcss/components/dialog.css:58]:
```suggestion
        @apply bg-black text-white w-full relative tracking-normal;
```

File: src/styles/tailwindcss/components/dialog.css [hunk 12]
```diff
@@ -56,8 +56,9 @@
...
-        @apply p-12 rounded-2xl;
+        @apply rounded-2xl;
         max-width: 470px;
+        padding: 30px;
         position: relative;
         letter-spacing: 0;
```
Comment (tiltom) [inline src/styles/tailwindcss/components/dialog.css:63]:
```suggestion
```

File: src/utils/helpers.ts [hunk 13]
```diff
@@ -75,3 +77,26 @@ export async function getWalletAddressForNetwork(networkType: NetworkType) {
   }
   return wallet.address;
 }
+
+export function detectWeb3Wallet() {
```
Comment (tiltom) [inline src/utils/helpers.ts:81]:
```suggestion
export const detectWeb3Wallet = () => {
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by creed-victor** [comment]
Spec changed so will continue working on this, but i think current files can be reviewed

**Review by ShynRou — CHANGES_REQUESTED** [review]
Testing rBTC to BTCS with one of my test addresses (0x653c65d44D062344A35a0204f94C1D0a59b54a25) leads to this error:
image
While using another the transfer worked, except that the bridge does not display the BTCS balance (on bscscan I can see that I received the tokens).

**Comment by creed-victor** [comment]
> Testing rBTC to BTCS with one of my test addresses (0x653c65d44D062344A35a0204f94C1D0a59b54a25) leads to this error: image While using another the transfer worked, except that the bridge does not display the BTCS balance (on bscscan I can see that I received the tokens).

@ShynRou, it always throws error for this address but not for other one?

**Comment by Alitasovryn** [comment]
Screen Shot 2022-04-07 at 8 02 18 AM
Screen Shot 2022-04-07 at 8 00 56 AM

Working fine

**Review by Alitasovryn — APPROVED** [review]

**Review by JoeStorm99 — APPROVED** [review]

**Pull request merged**